src/Core/Entity/Image/InputImage.php

```diff
@@ -38,6 +42,33 @@ public function __construct(OptionsBag $optionsBag, string $sourceImageUrl)
 
         $this->sourceImagePath = $optionsBag->hashOriginalImageUrl($this->sourceImageUrl);
         $this->saveToTemporaryFile();
+
+        // Store the source file mime type.
+        $this->sourceFileMimeType = finfo_file(
+            finfo_open(FILEINFO_MIME_TYPE),
+            $this->sourceImagePath
+        );
+
+        // If the source file has a video mime type, generate a full resolution
+        // image at the requested (or default) time duration and use that as
+        // the source image.
+        if (strpos($this->sourceFileMimeType, 'video/') !== false) {
+            $time = $this->getTime();
+            $tmpTime = str_replace(['.', ':'], '', $time);
+            $dest = $this->sourceImagePath . '-'. $tmpTime;
+            $overwrite = $this->optionsBag->get('refresh') ? ' -y' : ' -n';
+            $cmd = "ffmpeg " . $overwrite . " -i " . $this->sourceImagePath . " -vf scale='iw:ih' -ss " . $time .
+            " -f image2 -vframes 1 " . $dest . " -hide_banner -loglevel warning";
+            exec($cmd . ' 2>&1', $output);
```

Comment:
extending the `Processor` class will allow you to use the `execute` function which expects a `Command` argument

Reply:
See above.